### DISCUSSION TITLE
GDL on Ubuntu 20.04 on brand new Macbook Air with M1 chip - fantastic

### DISCUSSION THREAD

**Discussion by brandy125** [gnudatalanguage/gdl#910] [Show and tell]
I just installed GDL inside Ubuntu 20.04 on my brand new Macbook Air under Parallels.

The installation was straight forward without any problems, no complaints at all.

The speed is fantastic. A fft of a 10000x10000 float array performs in 1.6 seconds while it takes 6.4 seconds on my Mac mini with 3,2 GHz 6‑Core Intel Core i7 under IDL.

Even after compilation and some tests the Macbook is cold as my beer.

BTW,  compilation time took  5 min and 30 seconds using 1 CPU. Don't know if this is good or not. 

Here my configuration: 

Linux ubuntu 5.8.0-49-generic #55~20.04.1-Ubuntu SMP Fri Mar 26 01:00:41 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux

GDL - GNU Data Language, Version 1.0.0-rc.3 git

No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.2 LTS
Release:        20.04
Codename:       focal

Just wanted to post here in case that someone wants to try.

**Comment by GillesDuvert** [comment]
Hi @brandy125 , may I ask, is it 1000x10000 or 10000x10000 ?

**Reply by brandy125** [reply]
Hi @GillesDuvert, I just saw the typo and corrected it in my post.
It was an array of 10000 x 10000.

**Reply by alaingdl** [reply]
hi, thanks for the feedback, I am sure @GillesDuvert and @pjb7687 appreciate because they are in the middle of complex changes.

Concerning the compilation time, did you allocate more than one core to your U env. ? In such case you can use "make -j 2"
(or more, to be adjust). I do thank on VM and usually it approximately divided the total compilation time by the factor 
as long as it is below the number of total cores. (more tricks for fine tuning when SSD vs HDD, but it is another story)

**Reply by brandy125** [reply]
I allocated just one CPU. If I compile with "make -j8" it took 1m 20seconds.
We need to see that the M1 has 4 high performance cores and 4 high efficiency cores. Maybe only the high performance cores participated here.

**Reply by brandy125** [reply]
In general, should I report errors/problems with GDL on M1 and if yes, where should I post.
Or is it too early and does not make too much sense now?
For example I saw segmentation fault if I use window,/pixmap,/free before displaying an array but I do not know if it is worth to investigate further on it.

**Reply by slayoo** [reply]
please report any segmentation fault as a github issue, of course with as detailed as possible instructions on how to reproduce it (and if possible with a stack trace from your machine). Thanks!

**Comment by GillesDuvert** [comment]
**Then it is really impressive.** However (for the readers) a simple GDL fft uses [fftw](http://fftw.org/) and will be, indeed, very fast. But some options (/DIMENSION ?) will switch to another, ridiculously slow, code, as they were not ported to fftw. So your mileage may vary (and, help is welcome, as usual --- especially on this simple topic).
